cpp/benchmarks/synchronization/synchronization.h

```diff
@@ -1,104 +1,31 @@
 // Google Benchmark library
 #include <benchmark/benchmark.h>
 
-class CudaState;
-
-class CudaStateIterator {
-  private:
-    
-    benchmark::State::StateIterator state_iter;
-    
-    CudaState* parent;
-
-  public:
-    // The following is for operator* which is used in the range-based for loop.
-    // This is just a dummy convention to suppress compiler warnings.
-    struct __attribute__((unused)) Value {};
-    Value operator*() const { return Value(); }
-    
-    CudaStateIterator(CudaState* cuda_state, benchmark::State::StateIterator state_iter_): parent(cuda_state), state_iter(state_iter_) { }
-
-    CudaStateIterator& operator++();
-    
-    bool operator!=(CudaStateIterator const& cuda_state_iter);
-};
-
-class CudaState {
-  private:
-   
-    friend class CudaStateIterator;
-
-    cudaEvent_t start, stop;
-    bool started;
-    benchmark::State* p_state; 
+struct cuda_event_timer {
+  cudaEvent_t start, stop;
   
-  public:
+  benchmark::State* p_state;
 
-    CudaState(benchmark::State& state): p_state(&state), started(false) {
-      cudaEventCreate(&start);
-      cudaEventCreate(&stop);
-    }
+  cuda_event_timer(benchmark::State& state): p_state(&state) {
+    cudaEventCreate(&start);
+    cudaEventCreate(&stop);
+    cudaEventRecord(start);
+  }
+  
+  cuda_event_timer(){
```

Comment:
What's the point of this constructor? If you want to prevent default construction, you can just do this: 

```
cuda_event_timer() = delete;
```